controllers/discordactions.js

```diff
@@ -66,9 +67,25 @@ const createGroupRole = async (req, res) => {
 const getAllGroupRoles = async (req, res) => {
```

Comment:
here retrieveUsers function is expecting an array of userIds as input, but the code is passing a Set of userIds. This will cause the retrieveUsers function to throw an error.

To fix the bug, you need to change the line of code that calls the retrieveUsers function to the following:

const groupCreatorsDetails = await retrieveUsers({ userIds: Array.from(groupCreatorIds) });
This will convert the Set of userIds to an array, which is the correct format for the retrieveUsers function.